src/components/protocol_handler/src/handshake_handler.cc

```diff
@@ -72,6 +75,9 @@ bool HandshakeHandler::GetPolicyCertificateData(std::string& data) const {
 void HandshakeHandler::OnCertificateUpdateRequired() {}
 
 bool HandshakeHandler::OnHandshakeFailed() {
+  service_status_update_handler_->OnGetSystemTimeExpired(
```

Comment:
@mked-luxoft is that the only reason when `OnHandshakeFailed` called?

Reply:
@AKalinich-Luxoft  i think the only place where it is called is `BasicCommunicationGetSystemTimeRequest::onTimeOut`